rust/arrow/src/array.rs

```diff
@@ -971,6 +1034,11 @@ impl StructArray {
     pub fn column(&self, pos: usize) -> &ArrayRef {
         &self.boxed_fields[pos]
     }
+
+    /// Return the number of fields in this struct array
+    pub fn num_columns(&self) -> usize {
```

Comment:
Maybe num_children is a better name?

Reply:
This is to be consistent with the previous `column` method. `num_children` might be a little confusing.

I think a better term is `field` and `num_fields` because when we talk about struct we usually say "struct fields`. However this requires changing the current `column` method.